clang-tools-extra/clang-tidy/bugprone/TooSmallLoopVariableCheck.cpp

```diff
@@ -82,10 +82,14 @@ void TooSmallLoopVariableCheck::registerMatchers(MatchFinder *Finder) {
   // We are interested in only those cases when the loop bound is a variable
   // value (not const, enum, etc.).
   StatementMatcher LoopBoundMatcher =
-      expr(ignoringParenImpCasts(allOf(hasType(isInteger()),
-                                       unless(integerLiteral()),
-                                       unless(hasType(isConstQualified())),
-                                       unless(hasType(enumType())))))
+      expr(ignoringParenImpCasts(allOf(
+               hasType(isInteger()), unless(integerLiteral()),
+               unless(allOf(
+                   hasType(isConstQualified()),
+                   declRefExpr(to(varDecl(anyOf(
+                       hasInitializer(ignoringParenImpCasts(integerLiteral())),
+                       isConstexpr(), isConstinit())))))),
+               unless(hasType(enumType())))))
```

Comment:
Should I change it? if so please provide me with some more details.

Reply:
i were thinking, instead of:
```
unless(hasType(enumType()))
```
simply write:
```
unless(declRefExpr(to(enumConstantDecl()))),
unless(allOf(hasType(isConstQualified()),
                    declRefExpr(to(varDecl(anyOf(hasInitializer(ignoringParenImpCasts(declRefExpr(to(enumConstantDecl())))), isConstexpr(), isConstinit())))))
```
Simply to exclude only those that are enum constant and those that use const variable that is initialized with enum constant, you could merge this with exist changes, and reduce duplications.

Reply:
I see your point. Made the changes.

Reply:
I don't think the tests will pass with this change.

Reply:
If tests do not pass, then leave it, it can be changed under separate issue later.

Reply:
sure thing.